tawazi/_dag/dag.py

```diff
@@ -413,20 +451,92 @@ def setup(
             target_nodes=target_nodes, exclude_nodes=exclude_nodes, root_nodes=root_nodes
         )
 
-        # 3. remove setup nodes
+        # 3. remove non setup nodes
         graph.remove_nodes_from(
             [node_id for node_id in graph if node_id not in self.graph_ids.setup_nodes]
         )
+        return graph
 
-        # 4. execute the graph and set the results to setup_results
-        _, self.results, _ = execute(
-            exec_nodes=self.exec_nodes,
-            results=self.results,
-            active_nodes=self.actives,
-            max_concurrency=self.max_concurrency,
-            graph=graph,
+    def config_from_dict(self, config: Dict[str, Any]) -> None:
+        """Allows reconfiguring the parameters of the nodes from a dictionary.
+
+        Args:
+            config (Dict[str, Any]): the dictionary containing the config
+                example: {"nodes": {"a": {"priority": 3, "is_sequential": True}}, "max_concurrency": 3}
+
+        Raises:
+            ValueError: if two nodes are configured by the provided config (which is ambiguous)
+        """
+
+        def node_values(n: ExecNode, conf: Dict[str, Any]) -> Dict[str, Any]:
+            values = dataclasses.asdict(n)
```

Comment:
rename n to xn to adhere to the global naming

Reply:
also, why is that not a method of execnode ? I'd rather we don't have nested functions if they're note strictly necessary, this impedes readability

Reply:
correct. done for config from node.
For compose I will do it in future PRs when I will refactor it a little bit